docs/design-documents/20221024-stream-inspector.md

```diff
@@ -0,0 +1,87 @@
+# Stream Inspector
+
+## Goals
+
+Make troubleshooting pipelines easier by making it possible to inspect the data which is flowing through pipeline.
+
+## Requirements
+
+1. All pipeline components (accessible "from the outside", i.e. sources, processors and destinations) should be
+   inspectable.
+    * For v1, it's enough that destinations are inspectable.
+2. A source inspector should show the records coming out of the respective source.
+3. A destination inspector should show the records coming into the respective destination.
+4. A processor has two inspectors: one for the incoming data and one for the resulting data (i.e. the transformed
+   records).
+5. The inspector should provide the data about the component being inspected, from the time the inspection started.
+   Historical data is not required.
+6. The inspector should have a minimal footprint on the resource usage (CPU, memory etc.)
+7. It should be possible to specify which data is shown.
```

Comment:
What options do you have in mind here?

Reply:
At the moment, I'm mainly interested in seeing if there's enough interest in this. This way or the other, it should be possible, because you can always (and pretty easily) extract the data you want from the API response.

Reply:
I'll argue we should leave this out of the scope of this issue. It sounds like it's a matter of how the data is represented to the user, which should be handled on the client side IMO. The only use case where I see value in this is if the payload is soo huge you wouldn't want to send it all to the client, but that's probably an uncommon use case. I'd keep it simple for now.

Reply:
Agreed :+1: